internal/engine/build_errors.go

```diff
@@ -0,0 +1,68 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/windmilleng/tilt/internal/model"
+)
+
+// Nothing is on fire, this is an expected case like a container builder being
+// passed a build with no attached container. Don't need to show this to a user.
+type ExpectedError struct {
```

Comment:
i feel like ExpectedError is way too generic a name for this

maybe call this something like `ExpectationFailed` or `RedirectToNextBuilder`?

fwiw, i often find https://httpstatusdogs.com/ helpful for thinking thru taxonomies of errors. in this case, this is more like a 3xx (redirect) or 4xx (request can't be satisfied for some reason)